### PR TITLE
Modifying orienteering planner related msgs and srvs

### PR DESCRIPTION
**Pull request by SwonGao**
Hi, I have modified OP related services and messages, please take a look

### CODE REVIEW

File: msg/Defect.msg [hunk 1]
```diff
@@ -1,10 +1,8 @@
 # DISCLAIMER: this will not be the final messagge to describe a defect.
 # This is a temporary placeholder for demonstration purposes.
 
-# Severity levels
-# 1 = class A
-# 2 = class B
-# 3 = class C
+uint8 number
```
Comment (matteodv99tn) [inline msg/Defect.msg]: Why did you add the `number` here?
I don't think we shall hard-code an ID of the defect.
Reading the updated docs, you state that it is the order for the defect to be reworked, but I think that this concept is misleading.
After the you solve the orienteering, you provide a list of **ordered** defects, i.e., the plan that the robot shall follow.

File: msg/Defect.msg [hunk 2]
```diff
@@ -1,10 +1,8 @@
...
 uint8 severity
 geometry_msgs/Pose location
-float64 estimated_rework_time
+# Rework time is relevant to severity levels
+# float64 estimated_rework_time
```
Comment (matteodv99tn) [inline msg/Defect.msg]: If you don't need this, feel free to cancel it from here

File: msg/Defect.msg [hunk 3]
```diff
@@ -1,10 +1,8 @@
 # DISCLAIMER: this will not be the final messagge to describe a defect.
 # This is a temporary placeholder for demonstration purposes.
 
-# Severity levels
-# 1 = class A
-# 2 = class B
-# 3 = class C
+uint8 number
 uint8 severity
```
Comment (matteodv99tn) [inline msg/Defect.msg:6]: I see you also removed the enumeration values of the defect type, maybe because it's a repetition of the data contained in the documentation.

Maybe take a look [here](https://github.com/ros2/rosidl/issues/260), but it could be better if we explicitly write here in the message the different constants that are so attached to the message itself.
This can also make programming "safer", especially in C++.

**Comment by matteodv99tn** [comment]
Note that I slightly modified the CMakelists

File: msg/Defect.msg (hunk 1 above)
Comment (SwonGao) [reply msg/Defect.msg:4]: Yes, i think it is better to have one way to index nodes,  for me i think an ID would be helpful. Do you have any other suggestions?

Comment (SwonGao) [reply msg/Defect.msg]: > Why did you add the `number` here? I don't think we shall hard-code an ID of the defect. Reading the updated docs, you state that it is the order for the defect to be reworked, but I think that this concept is misleading. After the you solve the orienteering, you provide a list of **ordered** defects, i.e., the plan that the robot shall follow.

Yes, i think it is better to have one way to index nodes,  for me i think an ID would be helpful. Do you have any other suggestions?

Comment (matteodv99tn) [reply msg/Defect.msg]: In my opinion, the `number` index you are going to provide here will match exactly the index within `magician_msgs/Defect[] roadmap` vector that you will provide as output of the orienteering problem, so its a repetition.
Moreover, the "Defect" message could potentially be shared with other topics/services that might not strictly require an *indexing feature*.

If you really think that your message requires an ID, maybe create a new message (e.g. `OrienteeringDefect`) within a subfolder that will contains `magician_msgs/Defect data`, and everything else you need for the orienteering problem (that other services might not need)

**Comment by matteodv99tn** [comment]
Also, could it be useful to add location data within the `Defect` message? Like a `geometry_msgs/PoseStamped`

**Comment by SwonGao** [comment]
> Also, could it be useful to add location data within the `Defect` message? Like a `geometry_msgs/PoseStamped`

We already have 'geometry_msgs/Pose', as for the extra header in 'PoseStamped', i think we might not need timestamp for the 'Defect' message

File: msg/Defect.msg (hunk 3 above)
Comment (SwonGao) [reply msg/Defect.msg:6]: ok, I have added them back.

**Comment by matteodv99tn** [comment]
I'm sorry I completely overlooked the `location` parameter, my bad!

Maybe tomorrow I would get a look at the code and test few things on the constants, and hopefully we can close the pull request. 

See you tomorrow at the meeting

**Comment by matteodv99tn** [comment]
Hi,
I modified the `Defect.msg` to export the constants for the severity classes.
Like so we can have their values directly in our code linters and make the code more robust. This is a simple demo in C++:

```cpp
#include "magician_msgs/msg/defect.hpp"
// ...

int main(){
    // ...
    magician_msgs::msg::Defect msg;
    msg.severity = magician_msgs::msg::Defect::CLASS_A;
    // ...
}
```

Maybe we ask few things to Palopoli/Fontanelli at the meeting, and then you can merge

**Comment by SwonGao** [comment]
> Hi, I modified the `Defect.msg` to export the constants for the severity classes. Like so we can have their values directly in our code linters and make the code more robust. This is a simple demo in C++:
> 
> ```c++
> #include "magician_msgs/msg/defect.hpp"
> // ...
> 
> int main(){
>     // ...
>     magician_msgs::msg::Defect msg;
>     msg.severity = magician_msgs::msg::Defect::CLASS_A;
>     // ...
> }
> ```
> 
> Maybe we ask few things to Palopoli/Fontanelli at the meeting, and then you can merge

nice, thanks for the help, see you later!
